lib/doorkeeper/config.rb

```diff
@@ -392,5 +393,19 @@ def calculate_token_grant_types
       types << 'refresh_token' if refresh_token_enabled?
       types
     end
+
+    # Determine whether +reuse_access_token+ and +hash_secrets+
+    # have both been activated.
+    #
+    # In that case, disable reuse_access_token value and warn the user.
+    def validate_reuse_access_token_value
```

Comment:
Maybe we can move it in `reuse_access_token` setter ? It works once (on config build), so the behavior will be the same, but we wouldn't change the Configuration module API (no need to change `build` method)

Reply:
I initially tried that, but it will not work if you place `hash_secrets` after `reuse_access_token`. This will require the check at two places and that's why I moved it out of there. I can still move it in both setters if you prefer.